emu/src/cpu/arm7tdmi.rs

```diff
@@ -972,6 +999,12 @@ impl Arm7tdmi {
     }
 }
 
+/// Extends the bit at position `bit_to_extend` over a i32.
+const fn sign_extend(input: u32, bit_to_extend: u32) -> i32 {
+    let mask = 1 << bit_to_extend;
+    ((input as i32) ^ mask) - mask
+}
+
```

Comment:
I think the `bit_to_extend` is a bit misleading. For example if we have a 23 bits two-complements number that we want to extend, we should pass `22` as parameter because we want the mask to be 23 bits long (thus shifting `1` by 22). Maybe we could call the parameter `number_of_bits` or something like that and use `number_of_bits - 1` to create the mask and call the functions with the correct number (`23` in line `974` for example).